pkg/ddl/job_table.go

```diff
@@ -431,33 +374,77 @@ func (s *jobScheduler) checkAndUpdateClusterState(needUpdate bool) error {
 	return nil
 }
 
-func (s *jobScheduler) loadDDLJobAndRun(se *sess.Session) {
-	s.mu.RLock()
-	s.mu.hook.OnGetJobBefore()
-	s.mu.RUnlock()
+func (s *jobScheduler) loadAndDeliverJobs(se *sess.Session) error {
+	if s.generalDDLWorkerPool.available() == 0 && s.reorgWorkerPool.available() == 0 {
+		return nil
+	}
 
-	startTime := time.Now()
-	job, isReorg, err := s.getJob(se)
-	if job == nil || err != nil {
+	defer s.runningJobs.resetAllPending()
+
+	const getJobSQL = `select reorg, job_meta from mysql.tidb_ddl_job %s order by job_id`
+	var whereClause string
+	if ids := s.runningJobs.allIDs(); len(ids) > 0 {
+		whereClause = fmt.Sprintf("where job_id not in (%s)", ids)
+	}
+	sql := fmt.Sprintf(getJobSQL, whereClause)
+	rows, err := se.Execute(context.Background(), sql, "load_ddl_jobs")
```

Comment:
use `s.schCtx`?

Reply:
yes, this and below one are copied from previous impl